### ISSUE TITLE
Object Dependencies in SQL Server

### ISSUE THREAD

**Issue by dxps** [dbeaver/dbeaver#22815]
### Description

Dear all,

Being an old and happy DBeaver user, I am currently using it for interacting with an SQL Server.

I would like to see the references to an object (a table in this case) similar with what SQL Server Mgmt Studio shows as such:
![image](https://github.com/dbeaver/dbeaver/assets/14140226/b4aaf77c-0775-44cc-bc04-30f44d97699c)

In DBeaver, looking at the _References_ of the same object (table), nothing is shown:
![image](https://github.com/dbeaver/dbeaver/assets/14140226/3f2d55de-2b23-4a40-a222-cee9ac10ad76)

Is this is a bug, a missing feature or a misconfiguration :grey_question: 

Thanks in advance!

### DBeaver Version

Community 23.3.4

### Operating System

Pop!_OS 22.04 LTS

### Database and driver

1. Database version: Microsoft SQL Server 2019 (RTM-CU24) (KB5031908) - 15.0.4345.5 (X64)
2. Database driver: Microsoft JDBC Driver for SQL Server (MSSQL) ver. 12.4.1.jre11

![image](https://github.com/dbeaver/dbeaver/assets/14140226/54414d20-4fdc-4b93-aae8-4f566058f157)

**Comment by LonwoLonwo** [comment]
Hello @dxps 

What expectations do you have for the references tab? What kind of metadata do you expect to see there?

Now, the References folder shows FK connections with other tables like that:

![2024-02-14 10_05_00-Window](https://github.com/dbeaver/dbeaver/assets/45152336/af79e473-9ccc-4523-b84c-367cd58613f8)

So, maybe you are talking about a separate Dependencies folder? Like we have in PostgreSQL connections:

![2024-02-14 10_12_28-Window](https://github.com/dbeaver/dbeaver/assets/45152336/80d17171-442e-48ff-a496-d0560605f194)

**Comment by dxps** [comment]
Hi @LonwoLonwo,

That's precisely what I'd need: to have a (properly named, indeed) _Dependencies_ section as you shown above, but for SQL Server related objects.

Many thanks for the quick feedback! :pray:

**Comment by LonwoLonwo** [comment]
Ok, then I'll change the ticket status from "bug" to "future request".
And thanks for the idea.

**Comment by dr-1** [comment]
Agreed this would be nice to have. @dxps, for now you can get lists of dependencies, at least at a table level, with this query:
```
SELECT
	object_name(referencing_id) AS referencing_object,
	referenced_schema_name,
	referenced_entity_name
FROM sys.sql_expression_dependencies
```
Filter on the referencing object as needed.

**Comment by dxps** [comment]
@LonwoLonwo Thank you for the feedback and considering it! :pray: 

@dr-1 Thanks a lot, Dominik! :pray: That query is pretty handy!

For others that lands in this thread, a usage example of that query is as follows:

```sql
-- -----------------------------------------------
-- Get the objects that depend on a certain object
-- -----------------------------------------------

USE {my_db_name};

SELECT
    object_name(referencing_id) AS referencing_object,
    referenced_schema_name,
    referenced_entity_name
FROM sys.sql_expression_dependencies
where referenced_entity_name='{my_table_of_interest}'

-- ------------------------------------------------
-- Get the objects that a certain object depends on
-- ------------------------------------------------

USE {my_db_name};

SELECT
    object_name(referencing_id) AS referencing_object,
    referenced_schema_name,
    referenced_entity_name
FROM sys.sql_expression_dependencies
where object_name(referencing_id)='{my_table_of_interest}'

```
and replace `{my_table_of_interest}` with your object (such as table) name.

**Comment by dr-1** [comment]
@dxps, filtering on `referenced_entity_name` as in your example will show you the reverse dependencies, i.e. all the objects that depend on _this_ object. Which can be useful as well. To see which objects this one depends on, you'll want to filter on the _referencing_ entity (`WHERE object_name(referencing_id) = '{my_table_of_interest}'`).

**Comment by dxps** [comment]
@dr-1 You're absolutely right and thanks for the remark!

My main interest was to find out which objects depends on this one, as shown in the initial SSMS screenshot and also included below (although the title _Object Dependencies_ is a bit misleading, since you can get both its dependencies or the dependents on it).

![image](https://github.com/dbeaver/dbeaver/assets/14140226/3a7b8132-efda-443c-aa7f-e0d744ebf90d)

Updated my previous comment, so that we have the full picture in one place.

**Comment by jrbarkin** [comment]
I agree, this would be very useful!
